File: frontend/src/services/customtable.ts
import { apiClient } from "@/lib/api/vsentry-client";
import type { APIResponse } from "@/lib/api/vsentry-client";

export interface CustomTable {
  id?: number;
  ID?: number;
  created_at?: string;
  updated_at?: string;
  name: string;
  stream_fields: string;
  description?: string;
  query?: string;
  is_active?: boolean;
}

export const customTableService = {
  list: () => apiClient.get<any, APIResponse<CustomTable[]>>("/customtables/list"),
  
  add: (data: Partial<CustomTable>) => apiClient.post("/customtables/add", data),
  
  update: (data: Partial<CustomTable>) => apiClient.post("/customtables/update", data),
  
  delete: (id: number) => apiClient.post(`/customtables/delete?id=${id}`),
};

File: backend/model/customtable.go
package model

import "gorm.io/gorm"

type CustomTable struct {
	gorm.Model
	Name        string `json:"name"`        // Table - display
	StreamFields string `json:"stream_fields"` // _stream_fields - , e.g., "host,service"
	Description string `json:"description"` // Table - Query       string `json:"query"`       // Default - for this table
	IsActive    bool   `json:"is_active" gorm:"default:true"`
}